publications/tasks.py

```diff
@@ -1,27 +1,73 @@
 from django_q.models import Schedule
-from datetime import date, timezone
 from publications.models import Publication
-from django.core.mail import send_mail
+from views import harvest_data
+import requests
+from bs4 import BeautifulSoup
+import json
+import xml.dom.minidom
+from xml.dom.minidom import parseString
+from django.db.models import Max
+from django.contrib.gis.geos import GEOSGeometry
+
+def harvest_demo():
+    url1 = 'https://service.tib.eu/optimeta/index.php/optimeta/oai/?verb=ListRecords&metadataPrefix=oai_dc'
+    harvest_data(url1)
+
+def get_geom(url):
```

Comment:
If you add another funktion `parse_html` here that takes the content of the response, then you can more easily test the function. Then you have `get_geom` as a minimal wrapper that just has two lines: get the content and call `parse_..`.

Does that make sense to you?

This is independent of the option to of course mock the request in the test and test `geo_geom` as it is.

Reply:
yeah makes sense. Created a new function called `parse_html`. the new test is solely based on this fucntion